chain/network/src/broadcast/mod.rs

```diff
@@ -58,13 +58,17 @@ impl<T: Clone + Send> Receiver<T> {
 
     pub async fn recv(&mut self) -> T {
         self.next += 1;
-        let l = self.channel.stream.read().unwrap();
-        let n = self.channel.notify.notified();
-        let v = if l.len() > self.next - 1 { Some(l[self.next - 1].clone()) } else { None };
-        if let Some(v) = v {
-            return v;
-        }
-        drop(l);
+        let n = {
```

Comment:
please use more descriptive variable names ;-)
and add more comments on what this code does.